File: core/src/main/java/com/core/loitp/gallery/photos/PhotosDataCore.java
package com.core.loitp.gallery.photos;

import java.util.ArrayList;
import java.util.List;

import vn.loitp.restapi.flickr.model.photosetgetphotos.Photo;

/**
 * Created by www.muathu@gmail.com on 10/8/2017.
 */

public class PhotosDataCore {
    private final String TAG = PhotosDataCore.class.getSimpleName();
    private static final PhotosDataCore ourInstance = new PhotosDataCore();

    public static PhotosDataCore getInstance() {
        return ourInstance;
    }

    private PhotosDataCore() {
    }

    private List<Photo> photoList = new ArrayList<>();

    public List<Photo> getPhotoList() {
        return photoList;
    }

    public void setPhotoList(final List<Photo> photoList) {
        this.photoList = photoList;
        //LLog.d(TAG, "size: " + this.photoList.size());
    }

    public void addPhoto(final List<Photo> photoList) {
        if (this.photoList.isEmpty()) {
            setPhotoList(photoList);
            return;
        }
        if (photoList.size() > 300) {
            this.photoList.clear();
        }
        this.photoList.addAll(photoList);
        //LLog.d(TAG, "size: " + this.photoList.size());
    }

    public void clearData() {
        this.photoList.clear();
        //LLog.d(TAG, "size: " + this.photoList.size());
    }

    public int getSize() {
        if (this.photoList == null || this.photoList.isEmpty()) {
            return 0;
        }
        return this.photoList.size();
    }

    public Photo getPhoto(final int position) {
        if (this.photoList == null || this.photoList.isEmpty() || position < 0 || position > this.photoList.size()) {
            return null;
        }
        return this.photoList.get(position);
    }

    public int getPosition(final String photoID) {
        for (int i = 0; i < this.photoList.size(); i++) {
            if (photoID.equals(this.photoList.get(i).getId())) {
                return i;
            }
        }
        return -1;
    }
}

File: core/src/main/java/com/core/helper/gallery/photos/PhotosDataCore.kt
package com.core.helper.gallery.photos

import com.restapi.flickr.model.photosetgetphotos.Photo
import java.util.*

/**
 * Created by www.muathu@gmail.com on 10/8/2017.
 */
class PhotosDataCore private constructor() {
    private val logTag = PhotosDataCore::class.java.simpleName
    private var photoList = ArrayList<Photo>()

    companion object {
        val instance = PhotosDataCore()
    }

    fun getPhotoList(): List<Photo> {
        return photoList
    }

    fun setPhotoList(photoList: ArrayList<Photo>) {
        this.photoList = photoList
    }

    fun addPhoto(photoList: ArrayList<Photo>) {
        if (this.photoList.isEmpty()) {
            setPhotoList(photoList)
            return
        }
        if (photoList.size > 300) {
            this.photoList.clear()
        }
        this.photoList.addAll(photoList)
    }

    fun clearData() {
        photoList.clear()
    }

    val size: Int
        get() = if (photoList.isEmpty()) {
            0
        } else photoList.size

    fun getPhoto(position: Int): Photo? {
        return if (photoList.isEmpty() || position < 0 || position > photoList.size) {
            null
        } else photoList[position]
    }

    fun getPosition(photoID: String): Int {
        for (i in photoList.indices) {
            if (photoID == photoList[i].id) {
                return i
            }
        }
        return -1
    }
}
